algorithms/heap/fibonacci_heap.py

```diff
@@ -112,9 +112,15 @@ def merge(self, heap2):
         at the end of this heap's root list and connecting the heads 
         and tails.
         """
-        tail = self.root_list.left
+        if heap2.root_list == None:
+            return
         heap2_tail = heap2.root_list.left
-
+        if self.root_list == None:
+            self.root_list = heap2.root_list
+            self.min_node = heap2.min_node
+            return
+        tail = self.root_list.left
+        
```

Comment:
Flytta heap_tail under if satserna. Onödigt att allokera en variabel om den inte används i det fall vi går in i andra if-satsen.